lib/integrations/awsoidc/listdatabases.go

```diff
@@ -0,0 +1,314 @@
+		describeDBInstanceInput.Marker = &req.NextToken
+	}
+
+	rdsDBs, err := clt.DescribeDBInstances(ctx, describeDBInstanceInput)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	ret := &ListDatabasesResponse{}
+
+	if rdsDBs.Marker != nil && *rdsDBs.Marker != "" {
+		ret.NextToken = *rdsDBs.Marker
+	}
+
+	ret.Databases = make([]AWSDatabase, 0, len(rdsDBs.DBInstances))
+	for _, db := range rdsDBs.DBInstances {
+		awsDB, err := convertDBInstanceToDatabase(db)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		ret.Databases = append(ret.Databases, *awsDB)
+	}
+
+	return ret, nil
+}
+
+// convertDBInstanceToDatabase converts a Database from its rds/types.DBInstance representation into
+// a Teleport representation.
+func convertDBInstanceToDatabase(in types.DBInstance) (*AWSDatabase, error) {
+	ret := &AWSDatabase{}
+
+	if in.DBInstanceIdentifier == nil || *in.DBInstanceIdentifier == "" {
+		return nil, trace.BadParameter("database identifier not present")
+	}
+	ret.Name = *in.DBInstanceIdentifier
+
+	if in.DBInstanceStatus == nil || *in.DBInstanceStatus == "" {
+		return nil, trace.BadParameter("database status not present")
+	}
+	ret.Status = *in.DBInstanceStatus
+
+	if in.Engine == nil || *in.Engine == "" {
+		return nil, trace.BadParameter("database engine not present")
+	}
+	ret.Engine = *in.Engine
+
+	ret.Labels = convertTagListToLabels(in.TagList)
+
+	// Endpoint may not exist.
+	// This is the case when the Database was just created created but is not yet available.
+	if in.Endpoint != nil && in.Endpoint.Address != nil && *in.Endpoint.Address != "" && in.Endpoint.Port != 0 {
+		ret.Endpoint = fmt.Sprintf("%s:%d", *in.Endpoint.Address, in.Endpoint.Port)
+	}
+
+	if in.DbiResourceId == nil {
+		return nil, trace.BadParameter("database resourceid not present")
+	}
+	ret.ResourceID = *in.DbiResourceId
+
+	return ret, nil
+}
+
+func listDBClusters(ctx context.Context, clt ListDatabasesClient, req ListDatabasesRequest) (*ListDatabasesResponse, error) {
+	describeDBClusterInput := &rds.DescribeDBClustersInput{
+		Filters: []types.Filter{
+			{Name: &filterEngine, Values: req.Engines},
+		},
+	}
+	if req.NextToken != "" {
+		describeDBClusterInput.Marker = &req.NextToken
+	}
+
+	rdsDBs, err := clt.DescribeDBClusters(ctx, describeDBClusterInput)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	ret := &ListDatabasesResponse{}
+
+	if rdsDBs.Marker != nil && *rdsDBs.Marker != "" {
+		ret.NextToken = *rdsDBs.Marker
+	}
+
+	ret.Databases = make([]AWSDatabase, 0, len(rdsDBs.DBClusters))
+	for _, db := range rdsDBs.DBClusters {
+		awsDB, err := convertDBClusterToDatabase(db)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		ret.Databases = append(ret.Databases, *awsDB)
+	}
+
+	return ret, nil
+}
+
+// convertDBClusterToDatabase converts a Database from its rds/types.DBCluster representation into
+// a Teleport representation.
+func convertDBClusterToDatabase(in types.DBCluster) (*AWSDatabase, error) {
```

Comment:
Same question as above, can we just reuse existing "db cluster to database resource" we use in auto-discovery.

Reply:
See comment above